sudospawner/spawner.py

```diff
@@ -43,9 +43,15 @@ def do(self, action, **kwargs):
         yield p.stdin.write(json.dumps(kwargs).encode('utf8'))
         p.stdin.close()
         data = yield p.stdout.read_until_close()
+        # Read again if output is not json
+        if not data.decode('utf8').startswith('{'):
+            data = yield p.stdout.read_until_close()
```

Comment:
How is it that read_until_close can be called twice? Shouldn't this always return nothing since the stream is closed after it returns?